### PR TITLE
feat: add methods for v4.1 - OKTA-340176

### PR DESCRIPTION
**Pull request by shuowu**
Changes:
* adds `sdk.isAuthenticated` to resolve with non-pending `isAuthenticated` state
* changes `sdk.getAccessToken` and `getIdToken` to read from in-memory authState
* provides tokens to `sdk.getUser` from in-memory authState
* add `util.toRelativeUrl`

### CODE REVIEW

File: lib/browser/browser.ts [hunk 1]
```diff
@@ -382,26 +382,46 @@ class OktaAuthBrowser extends OktaAuthBase implements OktaAuth, SignoutAPI {
...
+        clearTimeout(timeoutId);
+      };
+      handler = ({isAuthenticated, isPending}) => {
+        if (!isPending) {
+          resolve(isAuthenticated);
+          clear();
+        }
+      };
+      timeoutId = setTimeout(() => {
+        resolve(false);
```
Comment (aarongranick-okta) [inline lib/browser/browser.ts:404]: should we clear before resolve?

File: lib/util.ts [hunk 2]
```diff
@@ -82,6 +82,14 @@ export function toAbsoluteUrl(url = '', baseUrl) {
   return url[0] === '/' ? `${baseUrl}${url}` : `${baseUrl}/${url}`;
 }
 
+export function toRelativeUrl(url = '', baseUrl) {
```
Comment (aarongranick-okta) [inline lib/util.ts:85]: where is this function being used?

File: lib/browser/browser.ts (hunk 1 above)
Comment (shuowu) [reply lib/browser/browser.ts:404]: either way should work, resolve won't stop the execution of the code after it.

File: lib/util.ts (hunk 2 above)
Comment (shuowu) [reply lib/util.ts:85]: It's mostly for the downstream SDKs, as they both depend on auth-js, looks like here can be a good place for some common util functions.

File: lib/browser/browser.ts [hunk 3]
```diff
@@ -382,26 +382,46 @@ class OktaAuthBrowser extends OktaAuthBase implements OktaAuth, SignoutAPI {
   // Common Methods from downstream SDKs
   //
 
+  async isAuthenticated(timeout?: number): Promise<boolean> {
```
Comment (swiftone) [inline lib/browser/browser.ts:385]: should this be `timeoutInMs`?

**Review by swiftone — COMMENTED** [review]
LGTM, will let Aaron approve since he had some questions

File: lib/browser/browser.ts (hunk 1 above)
Comment (aarongranick-okta) [reply lib/browser/browser.ts:404]: unless it throws?

**Review by aarongranick-okta — APPROVED** [review]

**Pull request merged**